### PR TITLE
Finished Task3 Add a GitHub action that runs Pytests on PRs in the project

### LINKED ISSUE #4: Add a GitHub action that runs Pytests on PRs in the project

### PR DESCRIPTION
**Pull request by Namra-Jabeen**
In order to achieve this, I have done the following:
1. Created a .yml file in .github/workflows. 
- The name of the workflow is CI. It is run on pull requests for all branches. Workflow_dispatch means that this action can be run manually.
- Then we add the steps to run the pytests: Set up Python 3.7. Install Poetry, Install dependencies specified in poetry.lock and pyproject.toml, and finally run Pytests in Task3/tests/ directory.
2. Moved poetry.lock and pyproject.toml to the root directory. 
- In order to install the poetry dependencies, and build the VM, poetry.lock and pyproject.toml has to be in root directory.
3. Created example tests in test_example_square.py
- This was to give pytests some initial tests to run. 
4. Created a README.md file where I describe how to add more tests to be run in this workflow, that is run on all PRs.

NOTE: There are many commit messages, because I have had to commit things to GitHub in order to try to see if the Github Action would build successfully.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -39,7 +39,7 @@ MANIFEST
 pip-log.txt
 pip-delete-this-directory.txt
```
Comment (halvgaard) [inline .gitignore:41]:
```suggestion
.idea/
```

Comment (halvgaard) [reply .gitignore:41]: Must ignore editor files

**Review by halvgaard — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (Namra-Jabeen) [reply .gitignore:41]: It's already on line 134 in the .gitignore file. Do you want me to move it to line 41? I don't think it should make a difference.

Comment (halvgaard) [reply .gitignore:41]: Ahh OK. If it's there, why has the .idea folder been added in a commit?

Comment (Namra-Jabeen) [reply .gitignore:41]: and now it should be gone!

**Review by halvgaard — APPROVED** [review]

**Pull request merged**